### PR TITLE
[Clustering Model] Fix template to support clustering model

### PR DESCRIPTION
**Pull request by weiguoz**
1. TRAIN
```SQL
SELECT *
FROM iris.train
TRAIN sqlflow_models.DeepEmbeddingClusterModel
WITH
    model.n_clusters = 5,
    model.run_pretrain = false,
    model.pretrain_dims = [4,500,500,2000,10]
COLUMN sepal_length, sepal_width, petal_length, petal_width
LABEL class   -- unnecessary
INTO sqlflow_models.my_clustering_model;
```
Right now, `LABEL` is required in the `TRAIN` SQL which is not reasonable here. 
I filed [another issue](https://github.com/sql-machine-learning/sqlflow/issues/839) about this.

2. PREDICT
```SQL
SELECT *
FROM iris.test
PREDICT iris.predict.class
USING sqlflow_models.my_clustering_model;
```

### CODE REVIEW

**Comment by Yancey0623** [comment]
> model.run_pretrain = false,

Do we need to specify this field? What happened if `run_pretrain=True`?

**Review by Yancey0623 — COMMENTED** [review]
Can we add a unit test for the user-defined train loop?

**Comment by weiguoz** [comment]
> > model.run_pretrain = false,
> 
> Do we need to specify this field? What happened if `run_pretrain=True`?

This is just for demonstration.
The only field that should be specified is `model.pretrain_dims`. The SQL can be reduced to:

```SQL
SELECT *
FROM iris.train
TRAIN sqlflow_models.DeepEmbeddingClusterModel
WITH
    model.pretrain_dims = [4,500,500,2000,10]
COLUMN sepal_length, sepal_width, petal_length, petal_width
LABEL class   -- unnecessary
INTO sqlflow_models.my_clustering_model;
```

**Comment by BlackPoint-CX** [comment]
> > model.run_pretrain = false,
> 
> Do we need to specify this field? What happened if `run_pretrain=True`?

In this version, this field should be set as true. It should be used with `existed_pretrain_model` to load existed pre-train model in next version.

File: sql/codegen_test.go [hunk 1]
```diff
@@ -33,14 +33,27 @@ WITH
...
 	testPredictSelectIris = `
 SELECT *
 FROM iris.test
 predict iris.predict.class
 USING sqlflow_models.my_dnn_model;
+`
+	testClusteringTrain = testSelectIris + `
+TRAIN sqlflow_models.DeepEmbeddingClusterModel
+WITH
+  model.pretrain_dims = [4,500,500,2000,10]
```
Comment (BlackPoint-CX) [inline sql/codegen_test.go]: As we discussed, it can be inferred by parameter `feature_columns` and `n_clusters`. 
User will not be forced to input for `pretrain_dims` but `n_cluster` should be required.

File: sql/executor_test.go [hunk 2]
```diff
@@ -113,9 +113,13 @@ func TestExecutorTrainAndPredictDNN(t *testing.T) {
 	a.NotPanics(func() {
 		stream := runExtendedSQL(testTrainSelectIris, testDB, modelDir, nil)
 		a.True(goodStream(stream.ReadAll()))
-
 		stream = runExtendedSQL(testPredictSelectIris, testDB, modelDir, nil)
 		a.True(goodStream(stream.ReadAll()))
+
+		stream = runExtendedSQL(testClusteringTrain, testDB, modelDir, nil)
```
Comment (Yancey0623) [inline sql/executor_test.go]: Maybe we need to put this case into another test case, it's not DNN.

Comment (weiguoz) [reply sql/executor_test.go]: Done.

**Comment by Yancey0623** [comment]
Hi @weiguoz , this PR is failed on Hive tests, seems saving model into hive table would cast too much time, can use the local fs instead of saving the model on table to fix the CI.

SQLFLow save the model into hive table by the `INSERT SQL`, I think we need to optimize this in the future, save into the local csv file and upload to the HDFS just like [Hive DBWriter](https://github.com/sql-machine-learning/sqlflow/blob/develop/sql/python/sqlflow_submitter/db_writer/hive.py) .

**Review by Yancey0623 — APPROVED** [review]
LGTM
